### PR TITLE
RFC: arm64 support [ci-skip] [not for merge]

### PR DESCRIPTION
**Pull request by mor1**
This PR is not for merge but is the starting point for ARM64 support.
It has now successfully built a kernel and initrd that have booted a packet.net ARM64 box.

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Can we deduplicate `kernel/Dockerfile,Makefile,kernel_config` across all the platforms?

i.e.
* kernel/Dockerfile: universal Dockerfile, accepts specifying ARCH via build-arg
* Makefile: ditto via `make ARCH=...`
* kernel_config: common parts across all the archs
* kernel_config.{x86_64,arm64,...}: architecture-specific part. Concatinated to kernel_config in the build container

**Comment by rneugeba** [comment]
@AkihiroSuda hopefully today I'll tidy up the kernel build a little and that might make it easier to dedup

File: Makefile [hunk 1]
```diff
@@ -5,15 +5,18 @@ all: default
...
 GIT_COMMIT=$(shell git rev-list -1 HEAD)
 
-GO_COMPILE=linuxkit/go-compile:4513068d9a7e919e4ec42e2d7ee879ff5b95b7f5@sha256:bdfadbe3e4ec699ca45b67453662321ec270f2d1a1dbdbf09625776d3ebd68c5
+GO_COMPILE=mor1/go-compile:021f2dca9228cdd27cffc304aafae1b1f900bfc2
 
 MOBY?=bin/moby
 LINUXKIT?=bin/linuxkit
 GOOS=$(shell uname -s | tr '[:upper:]' '[:lower:]')
-GOARCH=amd64
+GOARCH=arm64
```
Comment (justincormack) [inline Makefile]: you should not be setting this unconditionally, you should use `uname` to find it

**Comment by zlim** [comment]
cc @arm64b

**Comment by justincormack** [comment]
Going to close this as parts of arm64 are now being merged in and this does not apply.

**Pull request closed without merging**